sklearn/decomposition/tests/test_nmf.py

```diff
@@ -227,24 +354,28 @@ def test_nmf_sparse_input(solver, alpha_W, alpha_H):
     H1 = est1.components_
     H2 = est2.components_
 
-    assert_array_almost_equal(W1, W2)
-    assert_array_almost_equal(H1, H2)
+    assert_allclose(W1, W2)
+    assert_allclose(H1, H2)
 
 
-def test_nmf_sparse_transform():
+@pytest.mark.parametrize(
+    ["Estimator", "solver"], [[NMF, "cd"], [NMF, "mu"], [MiniBatchNMF, "mu"]]
+)
+def test_nmf_sparse_transform(Estimator, solver):
     # Test that transform works on sparse data.  Issue #2124
     rng = np.random.mtrand.RandomState(42)
     A = np.abs(rng.randn(3, 2))
     A[1, 1] = 0
     A = csc_matrix(A)
 
-    for solver in ("cd", "mu"):
-        model = NMF(
-            solver=solver, random_state=0, n_components=2, max_iter=400, init="nndsvd"
-        )
-        A_fit_tr = model.fit_transform(A)
-        A_tr = model.transform(A)
-        assert_array_almost_equal(A_fit_tr, A_tr, decimal=1)
+    # FIXME : should be removed in 1.1
+    init = "nndsvd"
+    model = Estimator(
+        solver=solver, random_state=0, n_components=2, max_iter=400, init=init
+    )
+    A_fit_tr = model.fit_transform(A)
+    A_tr = model.transform(A)
+    assert_allclose(A_fit_tr, A_tr, atol=1e-1)
```

Comment:
Should the version number be increased?
```suggestion
    # FIXME : should be removed in 1.3
    init = "nndsvd"
    model = Estimator(
        solver=solver, random_state=0, n_components=2, max_iter=400, init=init
    )
    A_fit_tr = model.fit_transform(A)
    A_tr = model.transform(A)
    assert_allclose(A_fit_tr, A_tr, atol=1e-1)
```

Reply:
No, this is an independent change. It was finalized in main but I somehow forgot some in this PR. I removed them.